### PR TITLE
Upgrade Hadoop and Hive

### PR DESCRIPTION
**Pull request by brianhw**
Configuration Pull Request
---

* Upgrade Hadoop from 2.3.0 to 2.7.2.
* Upgrade Hive from 0.2.0 to 1.0.0.

Configuration changes:
* Disable datanucleus.autoCreateSchema.
* Instead, call Hive's schematool -initSchema

### CODE REVIEW

File: playbooks/roles/hive/templates/hive-site.xml.j2 [hunk 1]
```diff
@@ -23,6 +23,6 @@
   </property>
   <property>
     <name>hive.metastore.schema.verification</name>  
-    <value>true</value>
```
Comment (mulby) [inline playbooks/roles/hive/templates/hive-site.xml.j2]: Do we actually want this to be `true` but actually call the schema tool to initialize the schema?

That is, perhaps we want to remove the `autoCreateSchema` property and instead add an ansible task that does something like:

``` bash
schematool -dbType mysql -initSchema
```

**Review by mulby — APPROVED** [review]
Generally looks good (assuming it works)

We may want to consider an explicit call to the schematool.

**Comment by brianhw** [comment]
This has incorporated commented from Gabe's review of an earlier iteration.  I have built a box and tested that this works on some edx-analytics-pipeline acceptance tests.  The PR is ready for review again.  @mulby in particular, since I can't invite you to review a PR you have already approved.

**Review by azafty468 — APPROVED** [review]
👍 . It looks like hadoop switched from a release name to a hadoop name.  Is that correct?  Or is there a release version that is more appropriate for us to use?

**Comment by brianhw** [comment]
My understanding is that https://archive.apache.org/dist/hadoop/common/ is now where they keep all of their releases.   The old location only goes up to 2.5.0.

**Comment by mulby** [comment]
Still LGTM

**Review by feanil — APPROVED** [review]

**Pull request merged**